### PR TITLE
HBASE-25425 Some notes on RawCell

### PR DESCRIPTION
**Pull request by saintstack**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 46s |  master passed  |
| +1 :green_heart: |  compile  |   0m 48s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 35s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 42s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 45s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 33s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  2s |  hbase-client in the patch passed.  |
|  |   |  28m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2797 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0176a1a765ce 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 772350b26e |
| Default Java | AdoptOpenJDK-1.8.0_232-b09 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/testReport/ |
| Max. process+thread count | 346 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 12s |  master passed  |
| +1 :green_heart: |  compile  |   0m 54s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 50s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 46s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  6s |  hbase-client in the patch passed.  |
|  |   |  30m 26s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2797 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux bc94b3c15890 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 772350b26e |
| Default Java | AdoptOpenJDK-11.0.6+10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/testReport/ |
| Max. process+thread count | 297 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  4s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 53s |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 46s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 50s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  19m  6s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  spotbugs  |   1m 59s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2797 |
| Optional Tests | dupname asflicense spotbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 18bd21d9c11e 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 772350b26e |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-client U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2797/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=3.1.12 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by wchevreuil — APPROVED** [review]

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Mutation.java [hunk 1]
```diff
@@ -1000,25 +999,16 @@ public byte getTypeByte() {
...
-      if (cell instanceof RawCell) {
-        return ((RawCell) cell).getTags();
-      }
       return PrivateCellUtil.tagsIterator(cell);
     }
 
     @Override
     public byte[] cloneTags() {
-      if (cell instanceof RawCell) {
-        return ((RawCell) cell).cloneTags();
```
Comment (virajjasani) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/Mutation.java:1020]: Ok so `RawCell.cloneTags()` internally refers `PrivateCellUtil.cloneTags(cell)` only, hence the removal and same is applicable to above methods.

**Review by virajjasani — APPROVED** [review]
+1

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/Mutation.java (hunk 1 above)
Comment (saintstack) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/Mutation.java:1020]: Yes.

Trivial change. Just shrinking the use of RawCell so it more obvious it is only needed in one location. Its weird having RawCell out in the client module when internally it is just doing the PrivateCellUtil method anyways.

I think this circumstance came about innocently after going back over git history... This was just not noticed before this.

**Pull request merged**